src/metabase/warehouse_schema/models/field.clj

```diff
@@ -131,17 +136,45 @@
   (let [defaults {:display_name (humanization/name->human-readable-name (:name field))}]
     (merge defaults field)))
 
+(def field-user-settings
+  "Set of user-settable values for a Field"
+  #{:semantic_type :description :display_name :visibility_type :has_field_values :effective_type :coercion_strategy :fk_target_field_id
+    :caveats :points_of_interest :nfc_path :json_unfolding :settings})
+
+(defn- ensure-field-user-settings-exist-for-fk-target-field [field]
+  (let [q {:select [:id]
+           :from [:metabase_field]
+           :where [:and
+                   [:= :fk_target_field_id (:id field)]
+                   [:not [:in :id {:select [:field_id]
+                                   :from [:metabase_field_user_settings]}]]]}
+        sql (sql/format q :dialect (mdb/quoting-style (mdb/db-type)))]
+    (t2/insert! :model/FieldUserSettings
+                (map (fn [{:keys [id]}] {:field_id id})
+                     (t2/query sql)))))
+
+(defn- sync-user-settings [field]
+  ;; we transparently prevent updates that would override user-set values
+  (let [user-settings (t2/select-one :model/FieldUserSettings (:id field))
+        updated-field (merge field (u/select-keys-when user-settings :non-nil field-user-settings))]
+    (t2.protocols/with-current field updated-field)))
+
 (t2/define-before-update :model/Field
   [field]
-  (u/prog1 field
-    (when (false? (:active (t2/changes <>)))
-      (t2/update! :model/Field {:fk_target_field_id (:id field)} {:semantic_type      nil
-                                                                  :fk_target_field_id nil}))))
+  (when (false? (:active (t2/changes field)))
+    (ensure-field-user-settings-exist-for-fk-target-field field)
+    (let [k {:fk_target_field_id (:id field)}
+          upds {:semantic_type      nil
+                :fk_target_field_id nil}]
+      (t2/update! :model/Field k upds)
```

Comment:
if this is updating the `field` passed to `before-update` you can `assoc` the keys to change into the return value and that will add them to the update.

Reply:
it isn't, this is triggering updates to related fields (this is preexisting behavior)